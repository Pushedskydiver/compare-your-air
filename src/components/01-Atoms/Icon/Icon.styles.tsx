import styled, { css } from 'styled-components';
import { colors } from '../../../styles/colors.styles';
import spacing from '../../../styles/spacing.styles';

interface IconProps {
  search?: boolean,
  close?: boolean
}

const IconStyles = styled.svg<IconProps>`
  display: inline-block;
  vertical-align: middle;
  pointer-events: none;

  ${props => props.close && css`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    padding: ${spacing.padding / 1.5}px;
    fill: ${colors.dark};
  `}

  ${props => props.search && css`
    position: absolute;
    top: 50%;
    left: 0;
    width: 48px;
    height: 48px;
    padding: 8px;
    fill: ${colors.neutral};
    transform: translateY(-50%);
    z-index: 3;
  `}
`;

export default IconStyles;
