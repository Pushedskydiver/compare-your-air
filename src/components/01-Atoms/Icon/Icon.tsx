import React from 'react';
import IconStyles from './Icon.styles';

export function Search() {
  return (
    <IconStyles search xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M9.516 14.016c2.484 0 4.5-2.016 4.5-4.5s-2.016-4.5-4.5-4.5-4.5 2.016-4.5 4.5 2.016 4.5 4.5 4.5zm6 0l4.969 4.969-1.5 1.5-4.969-4.969v-.797l-.281-.281c-1.125.984-2.625 1.547-4.219 1.547C5.907 15.985 3 13.126 3 9.516S5.906 3 9.516 3s6.469 2.906 6.469 6.516c0 1.594-.563 3.094-1.547 4.219l.281.281h.797z" /></IconStyles>
  );
}

export function Close() {
  return (
    <IconStyles close xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M18.984 6.422L13.406 12l5.578 5.578-1.406 1.406L12 13.406l-5.578 5.578-1.406-1.406L10.594 12 5.016 6.422l1.406-1.406L12 10.594l5.578-5.578z" /></IconStyles>
  );
}
