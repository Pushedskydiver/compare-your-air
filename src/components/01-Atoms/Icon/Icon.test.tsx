import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Close, Search } from './Icon';

afterEach(() => {
  cleanup();
});

describe('<Icons />', () => {
  it('should match close icon snapshot', () => {
    const { container } = render(
      <Close />
    );

    expect(container).toMatchSnapshot();
  });

  it('should match search icon snapshot', () => {
    const { container } = render(
      <Search />
    );

    expect(container).toMatchSnapshot();
  });
})
