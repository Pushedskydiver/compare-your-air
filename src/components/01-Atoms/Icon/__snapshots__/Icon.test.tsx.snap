// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icons /> should match close icon snapshot 1`] = `
<div>
  <svg
    class="sc-AxjAm hRrsiY"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.984 6.422L13.406 12l5.578 5.578-1.406 1.406L12 13.406l-5.578 5.578-1.406-1.406L10.594 12 5.016 6.422l1.406-1.406L12 10.594l5.578-5.578z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should match search icon snapshot 1`] = `
<div>
  <svg
    class="sc-AxjAm hzCFGq"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.516 14.016c2.484 0 4.5-2.016 4.5-4.5s-2.016-4.5-4.5-4.5-4.5 2.016-4.5 4.5 2.016 4.5 4.5 4.5zm6 0l4.969 4.969-1.5 1.5-4.969-4.969v-.797l-.281-.281c-1.125.984-2.625 1.547-4.219 1.547C5.907 15.985 3 13.126 3 9.516S5.906 3 9.516 3s6.469 2.906 6.469 6.516c0 1.594-.563 3.094-1.547 4.219l.281.281h.797z"
    />
  </svg>
</div>
`;
